Sources/Common/Store/SdkConfig.swift

```diff
@@ -20,6 +20,19 @@ public struct SdkConfig {
         public static func create(region: Region) -> SdkConfig {
             SdkConfig(trackingApiUrl: region.productionTrackingUrl)
         }
+
+        public static func create(region: Region, params: [String: Any]) -> SdkConfig {
+            var config = SdkConfig(trackingApiUrl: region.productionTrackingUrl)
+            config.autoTrackDeviceAttributes = params[CustomerIOBuilderConfigKeys.Config.autoTrackDeviceAttributes] as? Bool ?? config.autoTrackDeviceAttributes
```

Comment:
```suggestion
            if let autoTrackDeviceAttributes = params[CustomerIOBuilderConfigKeys.Config.autoTrackDeviceAttributes] as? Bool {
              config.autoTrackDeviceAttributes = autoTrackDeviceAttributes
            }
```

This suggestion is more of a swift design pattern. 

This idea can be applied to all of the lines here in this file. 

Also, I think it reads better then: `?? defaultValue` logic.

Reply:
Also, automated tests would be great to add for future maintainability of this function. 

Tests that would catch for us: If we ever add a new parameter to the `SdkConfig` object, a test automatically fails reminding us we need to update this function.

Reply:
Question for both of our maybe, but when accessing nested attributes of an object like CustomerIOBuilderConfigKeys.Config.autoTrackDeviceAttributes is some of the syntax that I don't yet understand here protecting against cases where CustomerIOBuilderConfigKeys or CustomerIOBuilderConfigKeys.Config might not exist for some reason?

Reply:
> Also, I think it reads better then: ?? defaultValue logic.

Agreed, still new to swift design pattern, but this looks better with it 👍

Reply:
>CustomerIOBuilderConfigKeys or CustomerIOBuilderConfigKeys.Config might not exist for some reason.

I don't completely understand the questions, but these constants are keys, we expect for the `map`. 
They are hardcoded constants and if the map dont have these constant keys, then we won't read its config value.